src/ralph/admin/widgets.py

```diff
@@ -1,21 +1,104 @@
 # -*- coding: utf-8 -*-
-
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
+import os
+from itertools import groupby
 
 from django import forms
-
 from django.contrib.admin.templatetags.admin_static import static
+from django.forms.utils import flatatt
+from django.template.defaultfilters import title, slugify
+from django.utils.safestring import mark_safe
+from django.utils.translation import ugettext as _
 
 
 class AdminDateWidget(forms.DateInput):
     @property
     def media(self):
         js = ['foundation-datepicker.js']
-        return forms.Media(js=[static('vendor/js/%s' % path) for path in js])
+        return forms.Media(js=[
+            static(os.path.join('vendor', 'js', path)) for path in js
+        ])
 
     def render(self, name, value, attrs=None):
         attrs['class'] = 'datepicker'
         return super(AdminDateWidget, self).render(name, value, attrs=attrs)
+
+
+class PermissionsSelectWidget(forms.Widget):
+    @property
+    def media(self):
+        js = ['multi.js']
+        return forms.Media(js=[
+            static(os.path.join('src', 'js', path)) for path in js
+        ])
+
+    def value_from_datadict(self, data, files, name):
+        values = data.get(name)
+        if not values:
+            return
+        return values.split(',')
+
+    def render(self, name, value, attrs=None, choices=()):
+        final_attrs = self.build_attrs(
+            attrs, type='hidden', name=name, value=','.join(map(str, value)),
+        )
+        return mark_safe(
+            '<a class="expand action-expand">Expand all</a>'
+            '<ul class="accordion" data-multi="{}" data-accordion>{}</ul>'
+            '<input{} />'.format(
+                name, self.render_options(choices, value), flatatt(final_attrs)
+            )
+        )
+
+    def render_all_option(self, slug):
+        return (
+            '<input class="select-all" id="id_all_{id_name}" type="checkbox">'
+            '<label for="id_all_{id_name}">{label}</label><br>'
+        ).format(id_name=slug, label=_('All'))
+
+    def render_option(self, selected_choices, option_value, option_label):
+        input_id = 'id_option_{}'.format(option_value)
+        attrs = {'id': input_id, 'type': 'checkbox', 'value': option_value}
+        if option_value in selected_choices:
+            attrs['checked'] = 'checked'
+        attrs = self.build_attrs(**attrs)
+        return '<input{}><label{}>{}</label>'.format(
+            flatatt(attrs), flatatt({'for': input_id}), option_label
+        )
+
+    def render_options(self, choices, selected_choices):
+        separator = ' | '
+        self.choices = sorted(
+            self.choices, key=lambda x: x[1].split(separator)[1]
+        )
+        grouped = groupby(self.choices, lambda x: x[1].split(separator)[1])
+        rendered_options = ''
+        for group in grouped:
+            items = list(group[1])
```

Comment:
could you create some pretty names for `group[0]` and `group[1]`? Ex. `group_key` and `group_choices`?